airflow/migrations/versions/0045_3_0_0_add_dag_versioning.py

```diff
@@ -0,0 +1,151 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+add dag versioning.
+
+Revision ID: 2b47dc6bc8df
+Revises: 5f57a45b8433
+Create Date: 2024-10-09 05:44:04.670984
+
+"""
+
+from __future__ import annotations
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy_utils import UUIDType
+
+from airflow.migrations.db_types import StringID
+from airflow.models.base import naming_convention
+from airflow.utils import timezone
+from airflow.utils.sqlalchemy import UtcDateTime
+
+# revision identifiers, used by Alembic.
+revision = "2b47dc6bc8df"
+down_revision = "5f57a45b8433"
+branch_labels = None
+depends_on = None
+airflow_version = "3.0.0"
+
+
+def upgrade():
+    """Apply add dag versioning."""
+    op.create_table(
+        "dag_version",
+        sa.Column("id", UUIDType(binary=False), nullable=False),
```

Comment:
You could re-use the following that I had to add for https://github.com/apache/airflow/pull/43243. 

https://github.com/apache/airflow/blob/51f9e835bf4e40e418f9f45a510a639068b08fa2/airflow/migrations/versions/0042_3_0_0_add_uuid_primary_key_to_task_instance_.py#L161-L167

Reply:
The `sqlalchemy-utils` provided it as the 'Python' side default, and I changed it in the ORM to use UUID7 as the Python side default. It's not server-side, which makes me wonder if we should depend on uuid6.uuid7 entirely instead of the custom uuid? I set use_binary=False, which is the same as using string, only that it's `String(32)`. Another difference is that in your implementation, as_uuid=False was set, which I think would make the field to be String in Postgres.

Reply:
Looking at the code again: https://github.com/kvesteri/sqlalchemy-utils/blob/baf53cd1a3e779fc127010543fed53cf4a97fe16/sqlalchemy_utils/types/uuid.py#L29. The default provided is an example. So, UUIDType() without provided any default would result in a null value. Users have to specify a default which `Column(UUIDType(binary=False), primary_key=True, default=uuid6.uuid7)` provides.